ui/app/lib/satellizer.js

```diff
@@ -0,0 +1,856 @@
+          removeItem: function(key) { return undefined; }
+        };
+
+        switch (type) {
+          case 'localStorage':
+          case 'sessionStorage':
+            if (type in window && window[type] !== null) {
+              return window[type];
+            } else {
+              console.warn('Warning: Storage type `' + type + '` is disabled or unavailable. Satellizer will not work correctly.');
+              return stub;
+            }
+
+          default:
+            console.error('Error: Unexpected storage type `' + type + '`');
+            return stub;
+        }
+      };
+
+      return {
+        get: function(key) { return storage(config.storage).getItem(key); },
+        set: function(key, value) { return storage(config.storage).setItem(key, value); },
+        remove: function(key) { return storage(config.storage).removeItem(key); }
+      };
+    }])
+    .factory('satellizer.interceptor', [
+      '$q',
+      'satellizer.config',
+      'satellizer.storage',
+      function($q, config, storage) {
+        var tokenName = config.tokenPrefix ? config.tokenPrefix + '_' + config.tokenName : config.tokenName;
+        return {
+          request: function(httpConfig) {
+            var token = storage.get(tokenName);
+            if (token && config.httpInterceptor) {
+              if (config.authHeader && config.authToken) {
+                token = config.authToken + ' ' + token;
+              }
+              httpConfig.headers[config.authHeader] = token;
+            }
+            return httpConfig;
+          },
+          responseError: function(response) {
+            return $q.reject(response);
+          }
+        };
+      }])
+    .config(['$httpProvider', function($httpProvider) {
+      $httpProvider.interceptors.push('satellizer.interceptor');
+    }]);
+
+})(window, window.angular);
+
+// Base64.js Polyfill (@davidchambers)
+(function() {
+  var object = typeof exports != 'undefined' ? exports : this;
+  var chars = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/=';
+
+  function InvalidCharacterError(message) {
+    this.message = message;
+  }
+
+  InvalidCharacterError.prototype = new Error;
+  InvalidCharacterError.prototype.name = 'InvalidCharacterError';
+
+  object.btoa || (
+    object.btoa = function(input) {
+      var str = String(input);
+      for (var block, charCode, idx = 0, map = chars, output = ''; str.charAt(idx | 0) || (map = '=', idx % 1); output += map.charAt(63 & block >> 8 - idx % 1 * 8)) {
+        charCode = str.charCodeAt(idx += 3 / 4);
+        if (charCode > 0xFF) {
+          throw new InvalidCharacterError("'btoa' failed: The string to be encoded contains characters outside of the Latin1 range.");
+        }
+        block = block << 8 | charCode;
+      }
+      return output;
+    });
+
+  object.atob || (
+    object.atob = function(input) {
+      var str = String(input).replace(/=+$/, '');
+      if (str.length % 4 == 1) {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Expected '===' and instead saw '=='.](https://www.codacy.com/app/HiP-App/HiPCMS/file/2279397488/issues/source?bid=3144114&fileBranchId=3144125#l848)